perfkitbenchmarker/publisher.py

```diff
@@ -736,3 +758,56 @@ def PublishSamples(self):
     for publisher in self.publishers:
       publisher.PublishSamples(self.samples)
     self.samples = []
+
+
+def RepublishJSONSamples(path):
+  """Read samples from a JSON file and re-export them.
+
+  Args:
+    path: the path to the JSON file.
+  """
+
+  samples = []
+  with open(path, 'r') as file:
+    for line in file:
+      try:
+        raw_dict = json.loads(line)
+        labels = raw_dict.pop('labels')
+        metadata = dict()
+        # labels is a comma-seprated list of key, value pairs
+        for pair in labels.split(','):
+          # Need to strip leading and trailing '|' from each pair
+          key, _, value = pair[1:-1].partition(':')
```

Comment:
Might I suggest this for loading samples?

```
samples = [json.loads(s) for s in file.read().split('\n') if s]
for sample in samples:
  # Chop '|' at the beginning and end of labels and split labels by '|,|'
  fields = sample['labels'][1:-1].split('|,|')
  # Turn the fields into a [[key, value], ...]
  key_values = [field.split(':', 1) for field in fields]
  sample['metadata'] = { k: v for k, v in key_values }
```

It's concise and very robust. Allows commas and even '|' in metadata values.